### PR TITLE
docs: for global rule

### LINKED ISSUE #2854: Missing the API documentation of global rule
Global rule is only mentioned in `doc/architecture-design.md`. We should add a new section in `doc/admin-api.md`.

### PR DESCRIPTION
**Pull request by spacewander**
Fix #2854

Signed-off-by: spacewander <spacewanderlzx@gmail.com>

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [x] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: doc/admin-api.md [hunk 1]
```diff
@@ -688,6 +689,31 @@ Config Example:
 }
 ```
 
+## Global Rule
+
+*API*：/apisix/admin/global_rules/{id}
+
+*Description*: Set plugins which run globally. Those plugins will be run before any Route/Service level plugins.
```
Comment (juzhiyuan) [inline doc/admin-api.md:696]: and Consumer?

File: doc/admin-api.md [hunk 2]
```diff
@@ -688,6 +689,31 @@ Config Example:
...
+
+> Request Methods：
+
+|Method      |Request URI|Request Body|Description        |
+|---------|-------------------------|--|------|
+|GET      |/apisix/admin/routes|NULL|Fetch resource list|
+|GET      |/apisix/admin/routes/{id}|NULL|Fetch resource|
```
Comment (juzhiyuan) [inline doc/admin-api.md:703]: `routes`?

File: doc/admin-api.md (hunk 1 above)
Comment (spacewander) [reply doc/admin-api.md:696]: We don't execute Consumer plugin. The plugin is used to detect which consumer is matched.

File: doc/admin-api.md (hunk 2 above)
Comment (spacewander) [reply doc/admin-api.md]: Fixed

File: doc/admin-api.md [hunk 3]
```diff
@@ -688,6 +689,31 @@ Config Example:
...
+
+> Request Methods：
+
+|Method      |Request URI|Request Body|Description        |
+|---------|-------------------------|--|------|
+|GET      |/apisix/admin/global_rules|NULL|Fetch resource list|
+|GET      |/apisix/admin/global_rules/{id}|NULL|Fetch resource|
+|PUT      |/apisix/admin/global_rules/{id}|{...}|Create resource by ID|
+|DELETE   |/apisix/admin/global_rules/{id}|NULL|Remove resource|
+|PATCH    |/apisix/admin/global_rules/{id}|{...}|Standard PATCH. Update some attributes of the existing Route, and other attributes not involved will remain as they are; if you want to delete an attribute, set the value of the attribute Set to null to delete; especially, when the value of the attribute is an array, the attribute will be updated in full|
```
Comment (tokers) [inline doc/admin-api.md:706]: `Update some attributes of the existing Route` should be `Update some attributes of the existing global_rule`.

File: doc/zh-cn/admin-api.md [hunk 4]
```diff
@@ -695,6 +696,31 @@ ssl 对象 json 配置内容：
...
+
+> 请求方法：
+
+|名字      |请求 uri|请求 body|说明        |
+|---------|-------------------------|--|------|
+|GET      |/apisix/admin/global_rules|无|获取资源列表|
+|GET      |/apisix/admin/global_rules/{id}|无|获取资源|
+|PUT      |/apisix/admin/global_rules/{id}|{...}|根据 id 创建资源|
+|DELETE   |/apisix/admin/global_rules/{id}|无|删除资源|
+|PATCH    |/apisix/admin/global_rules/{id}|{...}|标准 PATCH ，修改已有 Route 的部分属性，其他不涉及的属性会原样保留；如果你要删除某个属性，将该属性的值设置为null 即可删除；特别地，当需要修改属性的值为数组时，该属性将全量更新|
```
Comment (tokers) [inline doc/zh-cn/admin-api.md:713]: `修改已有 Route 的部分属性` should be `修改已有 Global rule 的部分属性`.

File: doc/zh-cn/admin-api.md [hunk 5]
```diff
@@ -695,6 +696,31 @@ ssl 对象 json 配置内容：
...
+|PATCH    |/apisix/admin/global_rules/{id}/{path}|{...}|SubPath PATCH，通过 {path} 指定 Route 要更新的属性，全量更新该属性的数据，其他不涉及的属性会原样保留。|
```
Comment (tokers) [inline doc/zh-cn/admin-api.md:714]: `指定 Route 要更新的属性 ` Should be `指定 Global rule 要更新的属性`.

File: doc/admin-api.md [hunk 6]
```diff
@@ -688,6 +689,31 @@ Config Example:
...
+|PATCH    |/apisix/admin/global_rules/{id}/{path}|{...}|SubPath PATCH, specify the attribute of Route to be updated through {path}, update the value of this attribute in full, and other attributes that are not involved will remain as they are.|
```
Comment (tokers) [inline doc/admin-api.md:707]: `specify the attribute of Route to be updated through ` should be `specify the attribute of Global rule to be updated through`.

File: doc/zh-cn/admin-api.md (hunk 5 above)
Comment (spacewander) [reply doc/zh-cn/admin-api.md]: Fixed

**Review by juzhiyuan — APPROVED** [review]

**Review by tokers — APPROVED** [review]

**Pull request merged**